bin/improver-weighted-blending

```diff
@@ -54,31 +131,39 @@ def main():
         'ChooseDefaultWeightsNonLinear plugins. Then apply these '
         'weights to the dataset using the BasicWeightedAverage plugin.'
-        ' Required for ChooseDefaultWeightsLinear: y0val and ONE '
-        'of slope, ynval. Required for '
-        'ChooseDefaultWeightsNonLinear: cval.')
-    parser.add_argument('wts_calc_method',
+        ' Required for ChooseDefaultWeightsLinear: y0val and ynval.'
+        ' Required for ChooseDefaultWeightsNonLinear: cval.'
+        ' Required for ChooseWeightsLinear with dict: wts_dict.'
+        ' Required for ChooseWeightsLinear with cube: wts_mask_constraint.')
+
+    parser.add_argument('--wts_calc_method',
                         metavar='WEIGHTS_CALCULATION_METHOD',
-                        choices=['linear', 'nonlinear'],
-                        help='Method to use to calculate weights used in '
-                             'blending. "linear": calculate linearly varying '
-                             'blending weights. "nonlinear": calculate '
-                             'blending weights that decrease exponentially '
-                             'with increasing blending coordinate.')
+                        choices=['linear', 'nonlinear', 'dict', 'mask'],
+                        default='linear', help='Method to use to calculate '
+                        'weights used in blending. "linear" (default): '
+                        'calculate linearly varying blending weights. '
+                        '"nonlinear": calculate blending weights that decrease'
+                        ' exponentially with increasing blending coordinate. '
+                        '"dict": calculate weights using a dictionary passed '
+                        'in as a command line argument.  "mask": calculate '
+                        'spatially varying weights from the mask on an input '
+                        'data cube.')
+
     parser.add_argument('coordinate', type=str,
                         metavar='COORDINATE_TO_AVERAGE_OVER',
                         help='The coordinate over which the blending '
                              'will be applied.')
-    parser.add_argument('--coord_exp_val', metavar='COORD_EXPECTED_VALUES',
-                        type=str,
-                        help='Optional string of expected coordinate points '
-                        'seperated by , e.g. "1496289600, 1496293200"')
     parser.add_argument('--coordinate_unit', metavar='UNIT_STRING',
                         default='hours since 1970-01-01 00:00:00',
-                        help='Units for time coordinate. Default= '
+                        help='Units for blending coordinate. Default= '
                              'hours since 1970-01-01 00:00:00')
     parser.add_argument('--calendar', metavar='CALENDAR',
                         help='Calendar for time coordinate. Default=gregorian')
+    parser.add_argument('--cycletime', metavar='CYCLETIME', type=str,
```

Comment:
It might be worth saying here that if this isn't provided it defaults to the latest  cycle time in the input cubes if blending over models using forecast period weights.